pkg/throttle/throttle.go

```diff
@@ -0,0 +1,18 @@
+package throttle
```

Comment:
This is super nitty, but I don't think `throttle` is quite the right name for this package. To me, a throttle is something that limits a continuous process, e.g., a liquid or gas. So rate-limiting basically, but this construction is not limiting the rate at which you can do something, it's limiting the maximum number of things you can do concurrently.

Maybe the package should be called `limiter` or something. Then there could be types like `limiter.Fixed`, `limiter.Dynamic`, `limiter.Bucket`, `limiter.LeakyBucket` etc.

Just a thought anyway.